client/my-sites/plans-grid/components/header-price.tsx

```diff
@@ -163,46 +170,53 @@ const PlanFeatures2023GridHeaderPrice = ( {
 								priceDisplayWrapperClassName="plans-grid-2023__html-price-display-wrapper"
 							/>
 						</>
-					) }
-					{ ! shouldShowIntroPricing && shouldShowDiscountedPrice && (
+					) : (
 						<>
-							<Badge className="plan-features-2023-grid__badge">
-								{ isPlanUpgradeCreditEligible
-									? translate( 'Credit applied' )
-									: translate( 'One time discount' ) }
-							</Badge>
-							<PricesGroup isLargeCurrency={ isLargeCurrency }>
-								<PlanPrice
-									currencyCode={ currencyCode }
-									rawPrice={ originalPrice.monthly }
-									displayPerMonthNotation={ false }
-									isLargeCurrency={ isLargeCurrency }
-									isSmallestUnit={ true }
-									priceDisplayWrapperClassName="plans-grid-2023__html-price-display-wrapper"
-									original
-								/>
-								<PlanPrice
-									currencyCode={ currencyCode }
-									rawPrice={ discountedPrice.monthly }
-									displayPerMonthNotation={ false }
-									isLargeCurrency={ isLargeCurrency }
-									isSmallestUnit={ true }
-									priceDisplayWrapperClassName="plans-grid-2023__html-price-display-wrapper"
-									discounted
-								/>
-							</PricesGroup>
+							{ shouldShowDiscountedPrice ? (
+								<>
+									<Badge className="plan-features-2023-grid__badge">
+										{ isPlanUpgradeCreditEligible
+											? translate( 'Credit applied' )
+											: translate( 'One time discount' ) }
+									</Badge>
+									<PricesGroup isLargeCurrency={ isLargeCurrency }>
+										<PlanPrice
+											currencyCode={ currencyCode }
+											rawPrice={ originalPrice.monthly }
+											displayPerMonthNotation={ false }
+											isLargeCurrency={ isLargeCurrency }
+											isSmallestUnit={ true }
+											priceDisplayWrapperClassName="plans-grid-2023__html-price-display-wrapper"
+											original
+										/>
+										<PlanPrice
+											currencyCode={ currencyCode }
+											rawPrice={ discountedPrice.monthly }
+											displayPerMonthNotation={ false }
+											isLargeCurrency={ isLargeCurrency }
+											isSmallestUnit={ true }
+											priceDisplayWrapperClassName="plans-grid-2023__html-price-display-wrapper"
+											discounted
+										/>
+									</PricesGroup>
+								</>
+							) : (
+								<>
+									{ isAnyVisibleGridPlanDiscounted && (
+										<Badge className="plan-features-2023-grid__badge" isHidden={ true }></Badge>
```

Comment:
This will only work for single-line badges, which should address the current situation.
 
An alternative and better solution might involve moving these badges to a separate row. I think that will be a much deeper refactor, especially with having completely separate implementations in Comparison & Features grids.